data/data/openstack/masters/main.tf

```diff
@@ -32,9 +32,49 @@ data "ignition_config" "master_ignition_config" {
   ]
 }
 
+resource "openstack_blockstorage_volume_v3" "master_volume" {
+  name = "${var.cluster_id}-master-${count.index}"
+  count = var.root_volume_size == 0 ? 0 : var.instance_count
+
+  size = var.root_volume_size
+  volume_type = var.root_volume_type
+  image_id = data.openstack_images_image_v2.masters_img.id
+}
+
+resource "openstack_compute_instance_v2" "master_conf_from_vol" {
```

Comment:
I'm not clear on how to do it, but it would be nice if this could happen via conditionals inside `master_conf`, instead of adding a new `master_conf_from_vol`.  As it stands, you'll also need to bump things like our current:

```
pkg/terraform/gather/openstack/ip.go:	mrs, err := terraform.LookupResource(tfs, "module.masters", "openstack_compute_instance_v2", "master_conf")
pkg/terraform/gather/openstack/ip.go:			errs = append(errs, errors.Wrapf(err, "no access_ip_v4 for master_conf.%d", idx))
```

to have gather work right when we take this path.

Reply:
Right, I didn't know how to make `block_device` optional inside the `openstack_compute_instance_v2` resource, so I duplicated the resource for both cases.
Any suggestion on how to do it more cleanly?

Reply:
Very good point about `pkg/terraform/gather/openstack/ip.go`. I missed that.